### ISSUE TITLE
Creating cluster with an existing IAM roles

### ISSUE THREAD

**Issue by sdouche** [kubernetes-incubator/kube-aws#53]
Hello,
At work, I can't create roles. I have forked the file stack-template.json but It would be cool to have the option built-in.

Thanks.

**Comment by mumoshu** [comment]
@sdouche Hi, thanks for using kube-aws :bow:
Did you mean to allow kube-aws to *just assign* existing, pre-created iam roles instead of automatically creating ones?

**Comment by sdouche** [comment]
Hi @mumoshu. You're right, I want tu assign existing roles and not creating new ones.  this is the change I use in the `stack-template.json`.

I Added parameters:
```
  "Parameters" : {
    "IAMInstanceProfileEtcd" : {
        "Type" : "String",
        "Default" : "infra-LbcK8sEtcdInstanceProfile-XXX",
        "Description" : "Enter the etcd role name."
        },
    "IAMInstanceProfileController" : {
        "Type" : "String",
        "Default" : "infra-LbcK8sMasterInstanceProfile-XXX",
        "Description" : "Enter the controller role name."
        },
    "IAMInstanceProfileWorker" : {
        "Type" : "String",
        "Default" : "infra-LbcK8sMinionInstanceProfile-XXX",
        "Description" : "Enter the worker role name."
        }
  },
```

And removed `IAMInstanceProfile*` and `IAMRole*`.

**Comment by mumoshu** [comment]
@sdouche Thanks for your confirmation!

I believe this is something good to be included in kube-aws.
Maybe users with IAM roles like the [`PowerUser` managed IAM role](http://docs.aws.amazon.com/directoryservice/latest/admin-guide/role_power_user.html) assigned would benefit from it.

I'm labeling this `open for pull request` to make it clear that I'm ready to review/merge a pull request  addressing this issue.

**Comment by Krylon360** [comment]
I think this would be better to have the API servers and Workers run as a IAM InstanceRole that is trusted to ec2.amazonaws.com.

This will allow the workers to run as a InstanceProfile; and enable the ec2metadata endpoint to be utilized.
example: the kube2iam project.
https://github.com/jtblin/kube2iam

**Comment by mumoshu** [comment]
Hi @Krylon360!
Your suggestion seems to have been implemented via #297.

**Comment by mumoshu** [comment]
@sdouche Hi, long time no see 😄 

Thanks to @Fsero's recent effort https://github.com/kubernetes-incubator/kube-aws/pull/607, `worker.nodePools[].iam.instanecProfile.arn` and `controller.iam.instanceProfil.arn` can be used to reuse existing instance profiles respectively for worker and controller nodes.
The settings are available since the latest rc of kube-aws v0.9.7-rc.1.
I'd appreciate it if you could confirm if it works for you!

I'm closing this issue as resolved but please feel free to reopen if necessary.
